src/shared/json-queue.c

```diff
@@ -105,8 +133,260 @@ cJSON * jqueue_next(file_queue * queue) {
+        /*
+            First, we copy the first item in groups, then the rest,
+            in case there is more than one
+        */
+        os_calloc(1, strlen(cJSON_GetArrayItem(json_object, 0)->valuestring) + 1, groups);
+        strcpy(groups, cJSON_GetArrayItem(json_object, 0)->valuestring);
+
+        for (i = 1; i < cJSON_GetArraySize(json_object); i++) {
+            os_realloc(groups, strlen(groups) + strlen(cJSON_GetArrayItem(json_object, i)->valuestring) + 2, groups);
+            strcat(groups, ",");
+            strcat(groups, cJSON_GetArrayItem(json_object, i)->valuestring);
+        }
+
+        os_strdup(groups, al_data->group);
+
+        os_free(groups);
+    }
+
+    // Level
+    json_object = cJSON_GetObjectItem(rule, "level");
+
+    if (json_object) {
+        al_data->level = json_object->valueint;
+    }
+
+    /* Syscheck */
+    syscheck = cJSON_GetObjectItem(al_json, "syscheck");
+
+    if (syscheck) {
+        // Path
+        json_object = cJSON_GetObjectItem(syscheck, "path");
+
+        if (json_object) {
+            os_strdup(json_object->valuestring, al_data->filename);
+        }
+
+        // User
+        json_object = cJSON_GetObjectItem(syscheck, "uname_after");
+
+        if (json_object) {
+            os_strdup(json_object->valuestring, al_data->user);
+        }
+    }
+
+    /* Srcip */
+    srcip = cJSON_GetObjectItem(al_json, "srcip");
+
+    if (srcip) {
+        os_strdup(srcip->valuestring, al_data->srcip);
+    }
+
+    /* Location */
+    location = cJSON_GetObjectItem(al_json, "location");
+
+    if (location) {
+        os_strdup(location->valuestring, al_data->location);
+    }
+
+    /* Full log */
+    full_log = cJSON_GetObjectItem(al_json, "full_log");
+
+    os_realloc(al_data->log, 2 * sizeof(char *), al_data->log);
+
+    if (full_log) {
+        os_strdup(full_log->valuestring, al_data->log[0]);
+    }
+    else {
+        char *full_log_json = cJSON_PrintUnformatted(al_json);
+
+        os_strdup(full_log_json, al_data->log[0]);
+
+        os_free(full_log_json);
+    }
+
+    /*
+        Because of the format of the full_log field in the json file, the entire log is stored in a single line,
+        meanwhile, in the log file, the full log is stored in several lines.
+        When freeing memory of an alert we try to free several lines, but in this case there is only one, this is
+        why the second element of this array is set to NULL, to prevent a segmentation fault.
+    */
+    al_data->log[1] = NULL;
+
+    /* Free memory */
+    cJSON_Delete(al_json);
+    al_json = NULL;
+
+    return al_data;
+}
+
+/* Read from monitored file in JSON format */
+alert_data *Read_JSON_Mon(file_queue *fileq, const struct tm *p, unsigned int timeout) {
+    unsigned int i = 0;
+    alert_data *al_data;
+
+    /* If the file queue is not available, try to access it */
+    if (!fileq->fp) {
+        if (Handle_JQueue(fileq, 0) != 1) {
```

Comment:
Why are sleeping 5 seconds by calling to `file_sleep()` every time `Handle_JQueue()` fails? As I can see, when `Handle_JQueue()` fails, `Read_JSON_Mon()` returns NULL, so the reading finishes here: https://github.com/wazuh/wazuh/blob/f1bcd70cd024599643472b46d4457abcfea34da9/src/shared/report_op.c#L491 and the report is shown.

Reply:
I think this was a copy/paste of the same function in `file-queue.c` file.
https://github.com/wazuh/wazuh/pull/4256/files#diff-399a48392f0369ec1b77cbfb563939dcR128
I removed the sleep in both places.